x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/group_stream.ts

```diff
@@ -83,73 +71,84 @@ export class GroupStream extends StreamActiveRecord<Streams.GroupStream.Definiti
     desiredState: State,
     startingState: State
   ): Promise<ValidationResult> {
-    if (this.isDeleted()) {
-      return { isValid: true, errors: [] };
-    }
-
     if (this._definition.name.startsWith('logs.')) {
       throw new StatusError('A group stream name can not start with [logs.]', 400);
```

Comment:
If I create a group stream with the name of an existing (not persisted yet) classic stream, the UI will only show the group stream. We should check if the name is already taken